pinc/maybe_mail.inc

```diff
@@ -4,6 +4,135 @@ include_once($relPath.'misc.inc'); // html_safe()
 include_once($relPath.'Project.inc');
 include_once($relPath.'User.inc');
 
+use PHPMailer\PHPMailer\PHPMailer;
+use PHPMailer\PHPMailer\SMTP;
+use PHPMailer\PHPMailer\Exception;
+
+function maybe_email($to, $subject, $message, $from = null, $reply_to = null)
+// If this is for real, send the message.
+// If we're testing, just report what would have been sent.
+// Optional $from and $reply_to arguments may either take the form
+// "user@example.com" or "Example User <user@example.com>"
+{
+    global $testing, $auto_email_addr, $site_name, $site_url;
+
+    $from = $from ?? $auto_email_addr;
+    $reply_to = $reply_to ?? $auto_email_addr;
+
+    $success = true;
+
+    // Append a standard footer to all emails sent out by the system.
+    if (!endswith($message, "\n")) {
+        $message .= "\n";
+    }
+
+    $message .= implode("\n", [
+        "",
+        "--  ",
+        "$site_name  ",
```

Comment:
We should add a comment here that the trailing spaces is to force the lines into multiple paragraphs, took me a bit to remember why you did it like this.